stdlib/runtime/string.gr

```diff
@@ -393,38 +400,48 @@ let rec heapValueToString = (ptr, extraIndents, toplevel) => {
     },
     t when t == Tags._GRAIN_ADT_HEAP_TAG => {
       // [ <value type tag>, <module_tag>, <type_tag>, <variant_tag>, <arity>, elts ... ]
-      let variantName = getVariantName(ptr)
-
-      if (variantName == -1n) {
-        "<enum value>"
+      let builtinVariantName = getBuiltinVariantName(ptr)
+      if (builtinVariantName != -1n) {
+        // Assumes that all builtin variants do not have inline record
+        // constructors; if this changes this should be changed as well
+        tupleVariantToString(
+          ptr,
+          WasmI32.toGrain(builtinVariantName),
+          extraIndents
+        )
       } else {
-        let variantName = WasmI32.toGrain(variantName): String
-        // Check if this is a list
-        if (isListVariant(variantName)) {
-          listToString(ptr, extraIndents)
+        let variantPtr = getVariantMetadata(ptr)
+        if (variantPtr == -1n) {
+          "<enum value>"
         } else {
-          let variantArity = WasmI32.load(ptr, 16n)
-          if (variantArity == 0n) {
-            variantName
+          let length = WasmI32.load(variantPtr, 12n)
+          let variantName = allocateString(length)
+          Memory.copy(variantName + 8n, variantPtr + 16n, length)
+          let variantName = WasmI32.toGrain(variantName): String
+          // Check if this is a list
+          if (isListVariant(variantName)) {
```

Comment:
Isn't the list variant built-in now? Or are we still waiting on that change?

Reply:
I believe we are still waiting on #1593